### ISSUE TITLE
Google OAuth 2.0 with IdentityResource fails

### ISSUE THREAD

**Issue by lawik** [team-alembic/ash_authentication#480]
The `identity_resource` option assumes there will be a `uid` field that is not nil.

Google's is nil in my tests.

Since the resource is defined by ash_authentication more than my app I don't know quite what to do here.

**Comment by lawik** [comment]
Changing this line to include "id" as an option seems to work:

https://github.com/team-alembic/ash_authentication/blob/486a3989d5164097867b5e517a65797de25db7a6/lib/ash_authentication/user_identity/upsert_identity_change.ex#L34

**Comment by lawik** [comment]
New error after the above fix:

```
%Ash.Error.Invalid.NoPrimaryAction{resource: Knowb.Accounts.User, type: :read, changeset: nil, query: nil, error_context: [], vars: [], path: [], stacktrace: #Stacktrace<>, class: :invalid}
```

**Comment by lawik** [comment]
The NoPrimaryAction had all the info I needed at least. A default action on the User resource got me through.

I have no idea if the uid/sub/id fix would be the right thing. But it seems like it would work.

**Comment by lawik** [comment]
https://github.com/team-alembic/ash_authentication/pull/481

**Comment by jimsynz** [comment]
Closed by #481
